### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by bsmithai**
Features:

- Test various programs for checkpointing and benchmark them
- Mass benchmark bash script
- Python script for data analysis and plotting
- db for data analysis and data keeping

Future improvements:
- Restore benchmarks (Almost done)
- Analysis for restore
- cmd for benchmarking
- Better file structure for analysis results
- More charts for analysis
- Refactor benchmark code for readability

### CODE REVIEW

**Comment by nravic** [comment]
Looks good so far! Couple of nits, but otherwise this is great stuff. Slacked you a couple of comments too

**Comment by nravic** [comment]
Oh one more thing I'd add is the CI skip functions so the test passes

File: .ipynb_checkpoints/Untitled-checkpoint.ipynb
[the remark this answers is not in the document]
Comment (bsmithai) [reply .ipynb_checkpoints/Untitled-checkpoint.ipynb:1]: I believe it's from a job.yml I ran way back when I first started playing with this. Will remove! Thanks!

File: test/benchmark_analysis.png
[the remark this answers is not in the document]
Comment (bsmithai) [reply test/benchmark_analysis.png:1]: This is an artifact, I think I will just get rid of it and fix this file to go into a temp

File: test/README.md [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
-Wrangled some live proc data to test against. Should consider using filesystem mock in the future, but this is a quick and dirty way to run some tests. Should also consider pruning these for the future. For reference: 
+
+Wrangled some live proc data to test against. Should consider using filesystem mock in the future, but this is a quick and dirty way to run some tests. Should also consider pruning these for the future. For reference:
 
 - `1266999` -> is a process spawned by running `jupyter notebook &` (useful for testing interactive programs & python restores)
 - `1227709` -> is a process spawned by running `./server -m models/7B/ggml-model-q4_0.bin -c 2048 &` (useful for testing servers & network restores)
+
+### `large_data_benchmark.sh`
+
+- chmod +x the script before running
```
Comment (nravic) [inline test/README.md:12]: no need for this, github persists permission change

**Review by nravic — APPROVED** [review]
lgtm! just make updates where needed to the README and you're good to merge

**Pull request merged**